grails-rundeck-data-shared/src/main/groovy/rundeck/data/util/ExecutionDataUtil.groovy

```diff
@@ -0,0 +1,34 @@
+package rundeck.data.util
+
+import org.rundeck.app.data.model.v1.execution.ExecutionData
+import rundeck.data.constants.ExecutionConstants
+
+class ExecutionDataUtil {
+
+    public static String getExecutionState(ExecutionData e) {
+        return e.cancelled ? ExecutionConstants.EXECUTION_ABORTED :
+                (null == e.dateCompleted && e.status == ExecutionConstants.EXECUTION_QUEUED) ? ExecutionConstants.EXECUTION_QUEUED :
+                        null != e.dateStarted && e.dateStarted.getTime() > System.currentTimeMillis() ? ExecutionConstants.EXECUTION_SCHEDULED :
+                                (null == e.dateCompleted && e.status!=ExecutionConstants.AVERAGE_DURATION_EXCEEDED) ? ExecutionConstants.EXECUTION_RUNNING :
+                                        (e.status == ExecutionConstants.AVERAGE_DURATION_EXCEEDED) ? ExecutionConstants.AVERAGE_DURATION_EXCEEDED:
+                                                (e.status in ['true', 'succeeded']) ? ExecutionConstants.EXECUTION_SUCCEEDED :
+                                                        e.cancelled ? ExecutionConstants.EXECUTION_ABORTED :
+                                                                e.willRetry ? ExecutionConstants.EXECUTION_FAILED_WITH_RETRY :
+                                                                        e.timedOut ? ExecutionConstants.EXECUTION_TIMEDOUT :
+                                                                                (e.status == 'missed') ? ExecutionConstants.EXECUTION_MISSED :
+                                                                                        (e.status in ['false', 'failed']) ? ExecutionConstants.EXECUTION_FAILED :
+                                                                                                isCustomStatusString(e.status) ? ExecutionConstants.EXECUTION_STATE_OTHER : e.status.toLowerCase()
```

Comment:
Could this be written in a swith-case logic? Too many nested ternary operator doesn't look good